.github/workflows/release.yml

```diff
@@ -0,0 +1,65 @@
+name: release
+
+on:
+  push:
+    tags:
+      - "v[0-9]+*"
+
+permissions:
+  contents: write
+
+jobs:
+  release:
+    runs-on: ubuntu-latest
+    container:
+      image: ruby:2.6
+    steps:
+      - uses: actions/checkout@v3
+      - uses: hashicorp/vault-action@v2.4.2
+        with:
+          url: ${{ secrets.VAULT_ADDR }}
+          method: approle
+          roleId: ${{ secrets.VAULT_ROLE_ID }}
+          secretId: ${{ secrets.VAULT_SECRET_ID }}
+          secrets: |
+            secret/apm-team/ci/apm-agent-ruby-rubygems-release apiKey | API_KEY ;
+      - name: RubyGems login
+        run: |
+          RUBY_HOME="${HOME}/.gem"
+          RUBY_CREDENTIALS_FILE="${RUBY_HOME}/credentials"
+          mkdir -p "${RUBY_HOME}"
+          echo '---' > "${RUBY_CREDENTIALS_FILE}"
+          echo ":rubygems_api_key: ${API_KEY}" >> ~/.gem/credentials
+          chmod 0600 "${RUBY_CREDENTIALS_FILE}"
+      - name: Install build system
+        run: gem install rake yard rspec
```

Comment:
Wonder whether this particular step should be moved to a script or a composite action within this project, I've seen it's used in three different places

Reply:
https://github.com/elastic/apm-agent-ruby/pull/1366/commits/d832575c79c4e51c19385604e28f0d9e223aeda3